### ISSUE TITLE
Xbox wireless controller (from xbox series X)  wont pair in Bazzite-deck-gnome: 39

### ISSUE THREAD

**Issue by GotBearHeart** [ublue-os/bazzite#575]
Hello, 
The xbox wireless controller can not be paired over bluetooth on an freshly installed Bazzite-deck-gnome.

Reproduction step:
1. press the xbox wireless controller "pairing button" till the xbox icon flashes rapidly (activated pairing mode)
2. in game mode or desktop mode, open "settings" app, navigate to "bluetooth" section
3. "xbox wireless controller" appears in the found device
4. click the device to start pairing
5. --> failure, after some time, the pairing is not successful without error popup --> xbox wireless controller device remains "not set up".

rpm-ostree output:
> `rpm-ostree status
> State: idle
> Deployments:
> ● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:latest
>                    Digest: sha256:b7c54d8286dac4571bf866d7ae274d6cb49682556db2651a101f1a9f17f7649d
>                   Version: 39.20231120.0 (2023-11-21T03:00:24Z)
>                 Initramfs: '-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf' 
> 
>   ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:39
>                    Digest: sha256:b7c54d8286dac4571bf866d7ae274d6cb49682556db2651a101f1a9f17f7649d
>                   Version: 39.20231120.0 (2023-11-21T03:00:24Z)
>                 Initramfs: '-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf' 
> `

**Comment by beefchimi** [comment]
I have the exact same issue, with the following exception:

- I first installed `38` a few months ago.
- Could pair my `bluetooth` mouse without issue.
  - `XBox Elite Wireless Controller Series 2` however could not pair.
- Controller was able to pair after following these instructions: https://github.com/ublue-os/bazzite/issues/299
- However, I reported on a separate issue relating to the controller disconnecting: https://github.com/ublue-os/bazzite/issues/418
- I managed to get a `Windows 10` PC to update the firmware of the controller _(now on version `5.17.3202.0)_
- After updating the firmware, I was no longer able to pair the controller, getting the same issue as you.
- I rebased to `v39` hoping that would solve the issue...
  - Sadly, still the same issue.

**Comment by GotBearHeart** [comment]
Today I stumbled upon a solution by trying to use command line method of pairing, instead of using settings UI:
Following [xpadneo github page](https://github.com/atar-axis/xpadneo)
### Connection
 `sudo bluetoothctl`
- `[bluetooth]# scan on`
- wait until all available devices are listed (otherwise it may be hard to identify which one is the gamepad)
- push the connect button on upper side of the gamepad, and hold it down until the light starts flashing fast
- wait for the gamepad to show up in bluetoothctl, remember the address (e.g. `C8:3F:26:XX:XX:XX`)
- `[bluetooth]# scan off` to stop scanning as it may interfere with properly pairing the controller
- `[bluetooth]# pair <MAC>`
- `[bluetooth]# trust <MAC>`
- `[bluetooth]# connect <MAC>` (should usually not be needed but there are [open bugs](https://github.com/atar-axis/xpadneo/issues/198))
- The `<MAC>` parameter is optional if the command line already shows the controller name

You know that everything works fine when you feel the gamepad rumble ;)

**Comment by beefchimi** [comment]
Sadly, this method did not work for me :sob: I simply get `Device <MAC> is not available`.

I believe I followed your instructions exactly. I've pasted the `Konsole` lines for reference, just in case. In the snippet below, I have replaced all `MAC` addresses with `<HIDDEN-ADDRESS>`. In the case of the `Xbox` address, I replaced it with `<HIDDEN-ADDRESS-I-WANT>`. But when typing the command, I did in fact use the correct address. I also removed a few lines I believe were unrelevant _(just lots of `[NEW]` address lines)_.

```
beefchimi@fedora:/var/home/beefchimi$ sudo bluetoothctl
Agent registered
[MX Master 3S M]# scan on
hci0 type 7 discovering on
Discovery started
[CHG] Controller <HIDDEN-ADDRESS> Discovering: yes
[NEW] Device <HIDDEN-ADDRESS>
hci0 type 7 discovering off
hci0 type 7 discovering on
hci0 type 7 discovering off
[DEL] Device <HIDDEN-ADDRESS>
hci0 type 7 discovering on
[CHG] Device <HIDDEN-ADDRESS> RSSI: 0xffffffb5 (-75)
hci0 type 7 discovering off
hci0 type 7 discovering on
[NEW] Device <HIDDEN-ADDRESS-I-WANT> Xbox Wireless Controller
[MX Master 3S M]# scan off
hci0 type 7 discovering off
Discovery stopped
[CHG] Device <HIDDEN-ADDRESS> RSSI is nil
[MX Master 3S M]# pair <HIDDEN-ADDRESS-I-WANT>
Device <HIDDEN-ADDRESS-I-WANT> not available
[MX Master 3S M]# trust <HIDDEN-ADDRESS-I-WANT>
Device <HIDDEN-ADDRESS-I-WANT> not available
[MX Master 3S M]# connect <HIDDEN-ADDRESS-I-WANT>
Device <HIDDEN-ADDRESS-I-WANT> not available
[MX Master 3S M]# 
```

**Comment by ParetoOptimalDev** [comment]
> I simply get Device <MAC> is not available.

@beefchimi Just hold the "connect" button on the top of your controller untilt he light flashes.

**Comment by JaCh0o** [comment]
Same here, I can connect controller in gamepad steam but still doesn't work, no keys are responding

**Comment by Razuuu** [comment]
Same here with "Xbox Wireless Controller"
It's connecting, displays in Steam but no button works. If turning the controller of and on it wont connect again

**Comment by aviatorcards** [comment]
Interesting. Swapped to bazzite over the weekend and I'm finding that people are telling me to use Windows to update a controller that works fine under KDE, but not Gnome. Even years later.  Is there seriously not another fix for this? Why does it work under KDE and not Gnome?

Rebased to KDE from Gnome and that actually did not work.

Installed Aurora 41 and the controller was picked up immediately after install. No updates to the OS. Controller remained conneted and recognized after update to 42. I feel like this is a Gnome specific problem as I've had this issue in the past. Figured a "gaming distribution" would handle that.

![Image](https://github.com/user-attachments/assets/66248b6c-deb9-46e0-b0ee-da24950fe273)

**Comment by xkhoax** [comment]
Why this happens:
Bazzite (and Linux in general) stores Bluetooth pairing info under /var/lib/bluetooth. If a previous pairing with the controller failed or broke, the system may silently ignore that device instead of showing it again in the scan. The GUI doesn’t show any error, it just “doesn’t see” the controller.

What fixed it for me:

Turn Bluetooth OFF

Delete the old controller data (this part uses the terminal)

Open a terminal

First, list what’s inside /var/lib/bluetooth:

**sudo ls /var/lib/bluetooth**

You should see a folder that looks like your controller MAC address, for example:

54:8D:5A:D8:1B:BF

That folder is the stored data for your controller. Remove it:

**sudo rm -rf /var/lib/bluetooth/54:8D:5A:D8:1B:BF**

(Replace 54:8D:5A:D8:1B:BF with whatever folder name you actually see.)

**Comment by ghaspias** [comment]
Running Bazzite (Plasma) 44 on a Lenovo Legion 5 15ach6h, with a Xbox Wireless Controller (Model 1914). I don't know the firmware revision, but it was never updated. In default configuration, the bluetooth pairing UI (desktop mode) does not show the controller (after putting it in pairing mode, fast flashing). Pairing via bluetoothctl worked, but the controller never stopped blinking slowly, and would disconnect/reconnect every few seconds.
The solution in my case was to change the "Privacy" option in /etc/bluetooth/main.conf to "network". It now completes the pairing in bluetoothctl (stops blinking) and stays connected.

**Comment by MalvoTheReal** [comment]
My system dont see my Xbox controller, but i found a way to fix it, the problem mainly was in Privacy setting at: /etc/bluetooth/main.conf and BLE mode on my usb bluetooth adapter, as Xbox controllers use BLE mode to pair. Here is what helps, i find that fix with chatgpt and ask him to make a tuto:

Xbox Controller not detected over Bluetooth on Bazzite/Linux — fix for CSR/Barrot 0a12:0001

I ran into an issue on Bazzite KDE where an Xbox Wireless Controller worked perfectly over USB and also connected normally over Bluetooth in Windows, but Bazzite/Linux could not detect it at all.

Bluetooth adapter:




ID 0a12:0001 Cambridge Silicon Radio, Ltd Bluetooth Dongle


However, bluetoothctl show reported manufacturer 2279, which means this is actually one of the Barrot-based adapters that identifies itself using the old CSR USB ID.

The following errors appeared during scanning:




org.bluez.Error.InProgress


and with btmgmt:




Unable to start discovery. status 0x0d (Invalid Parameters)


What fixed it

In my case, the problem was related to Bluetooth Privacy.

Open:




sudo nano /etc/bluetooth/main.conf


Under the [General] section, set:




Privacy = off


Important: if you previously had:




Privacy = network


replace it with off.

Save the file and restart Bluetooth:




sudo systemctl restart bluetooth


Then reset the Bluetooth controller settings with:




sudo btmgmt power off
sudo btmgmt privacy off
sudo btmgmt le on
sudo btmgmt bredr on
sudo btmgmt power on


Check the current state:




sudo btmgmt info


Under current settings, the regular privacy flag should no longer be present.

ll-privacy may still remain — that is fine.

Test Xbox Controller discovery

Put the Xbox Controller into pairing mode so the Xbox button is flashing rapidly.

Run an LE-only scan:




sudo btmgmt find -l


After applying the fix, my controller immediately appeared:




name Xbox Wireless Controller
type LE Public


So BLE discovery was finally working correctly.

Pair the controller

Stop the low-level scan:




sudo btmgmt stop-find


Then start:




bluetoothctl


Inside bluetoothctl, run:




power on
agent on
default-agent
scan le


Once the Xbox Controller appears, note its MAC address, for example:




68:6C:E6:59:1C:43


Then run:




pair 68:6C:E6:59:1C:43
trust 68:6C:E6:59:1C:43
connect 68:6C:E6:59:1C:43


You can verify the connection with:




info 68:6C:E6:59:1C:43


A successful result should look similar to:




Paired: yes
Bonded: yes
Trusted: yes
Connected: yes


After that, the Xbox Controller should work normally over Bluetooth in Bazzite/Steam.

Short version

If Bazzite/Linux detects your adapter as:




0a12:0001 Cambridge Silicon Radio


but cannot find your Xbox Controller, try:




# /etc/bluetooth/main.conf

[General]
Privacy = off


Then run:




sudo systemctl restart bluetooth

sudo btmgmt power off
sudo btmgmt privacy off
sudo btmgmt le on
sudo btmgmt bredr on
sudo btmgmt power on

sudo btmgmt find -l


In my case, disabling Bluetooth Privacy immediately fixed BLE scanning, and the Xbox Wireless Controller became visible and connected successfully.

One more note: run btmgmt commands with sudo. Without it, you may get:




status 0x14 (Permission Denied)


Also, if the controller is detected with a very weak RSSI such as -90 to -100, keep it close to the Bluetooth dongle during the first pairing attempt.
